modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/DistributedConfigurationUpdater.java

```diff
@@ -38,56 +33,31 @@ public class DistributedConfigurationUpdater implements IgniteComponent {
 
     private static final IgniteLogger LOG = Loggers.forClass(DistributedConfigurationUpdater.class);
 
+    private final CompletableFuture<ConfigurationPresentation<String>> clusterCfgPresentation = new CompletableFuture<>();
     private final CompletableFuture<SecurityConfiguration> securityConfigurationFuture = new CompletableFuture<>();
 
-    public void setClusterRestConfiguration(SecurityConfiguration securityConfiguration) {
-        securityConfigurationFuture.complete(securityConfiguration);
+    public void setDistributedConfigurationPresentation(ConfigurationPresentation<String> presentation) {
```

Comment:
This method looks very much like a cycle in dependencies.
Maybe we should think twice about how the code is organized. Right now it seems a little messy.